### PR TITLE
Disable elisp-lint by default (fixes #22)

### LINKED ISSUE #22: Disable elisp-lint by default
It almost never works good in a freshly initialized project. It has a bad configuration system and/or default config. Right now it reports 3298 warnings in eldev itself, most of which are caused by misconfiguration, and the rest are also detected by other linters. I think these reasons are enough to disable elisp-lint by default.

### PR DESCRIPTION
**Pull request by sirikid**

### CODE REVIEW

File: eldev.el [hunk 1]
```diff
@@ -3050,15 +3050,37 @@ error.")
...
 
 (defmacro eldev-deflinter (name arguments &rest body)
-  "Register a linter in Eldev."
+  "Register a linter in Eldev.
+
+BODY can contain the following keywords:
+
+    :name NAME
+
+        Cleaner name (a symbol) for ther command line.  Default
```
Comment (doublep) [inline eldev.el:3069]: Forgot to change to "Linter name".

File: eldev.el [hunk 2]
```diff
@@ -3327,6 +3356,7 @@ least one warning."
 also runs `package-lint', so you either shouldn't run both or
 change `elisp-lint's configuration."
   :aliases        (elisp-lint el)
+  :default        nil
```
Comment (doublep) [inline eldev.el:3359]: Just remove this line, it's fine to rely on `nil` being the default.

**Review by doublep — CHANGES_REQUESTED** [review]

File: eldev.el (hunk 1 above)
Comment (sirikid) [reply eldev.el:3069]: Lol

**Review by doublep — APPROVED** [review]

File: eldev.el [hunk 3]
```diff
@@ -3137,8 +3159,12 @@ least one warning."
       (eldev-print "Nothing to do"))))
 
 (defun eldev-lint-default-p (linter)
-  (and (or (eq eldev-lint-default t) (memq linter eldev-lint-default))
-       (not (memq linter eldev-lint-default-excluded)) (not (memq linter eldev-lint-disabled))))
+  (and (eldev-pcase-exhaustive eldev-lint-default
+         (`t t)
+         (:default (assq :default (assq linter eldev--linters)))
```
Comment (doublep) [inline eldev.el:3164]: As it stands, I get the following if I try running it:

    ~/eldev$ eldev lint
    Wrong type argument: listp, (elisp . eldev-linter-elisp)

**Review by doublep — CHANGES_REQUESTED** [review]

**Comment by sirikid** [comment]
I think I fixed it, but I didn't add the test.

**Review by doublep — APPROVED** [review]

**Comment by doublep** [comment]
> but I didn't add the test

Doesn't seem to be really needed here, so it's OK.

Seems to work fine now. Also doesn't break Cider.

Merged, thank you.

**Pull request merged**